pyat/at/collective/beam_loading.py

```diff
@@ -193,16 +210,25 @@ def clear_history(self, ring=None):
         self._init_bl_params(current)
 
     def _init_bl_params(self, current):
-        theta = -self._vcav[1]+numpy.pi/2
-        vb = 2*current*self.Rshunt
-        a = self.Voltage*numpy.cos(theta-self._phis)
-        b = self.Voltage*numpy.sin(theta-self._phis)-vb*numpy.cos(theta)
-        psi = numpy.arcsin(b/numpy.sqrt(a**2+b**2))
-        vgen = self.Voltage*numpy.cos(psi) + \
-            vb*numpy.cos(psi)*numpy.sin(self._phis)
-        self._vgen = numpy.array([vgen, psi])
+        if (self._cavitymode == 1) and (current > 0.0):
+            theta = -self._vcav[1]+numpy.pi/2
+            vb = 2*current*self.Rshunt
+            a = self.Voltage*numpy.cos(theta-self._phis)
+            b = self.Voltage*numpy.sin(theta-self._phis)-vb*numpy.cos(theta)
+            psi = numpy.arcsin(b/numpy.sqrt(a**2+b**2))
+            vgen = self.Voltage*numpy.cos(psi) + \
+                vb*numpy.cos(psi)*numpy.sin(self._phis)
+
+        elif (self._cavitymode == 1) and (current == 0.0):
```

Comment:
This should be an else at the end no?

Reply:
I put an else and added an AtError for an unkown cavity and current configuration

Reply:
Sorry I explained wrong, I was thinking of:

```
   def _init_bl_params(self, current):
        if (self._cavitymode == 1) and (current > 0.0):
            theta = -self._vcav[1]+numpy.pi/2
            vb = 2*current*self.Rshunt
            a = self.Voltage*numpy.cos(theta-self._phis)
            b = self.Voltage*numpy.sin(theta-self._phis)-vb*numpy.cos(theta)
            psi = numpy.arcsin(b/numpy.sqrt(a**2+b**2))
            vgen = self.Voltage*numpy.cos(psi) + \
                vb*numpy.cos(psi)*numpy.sin(self._phis)
        elif self._cavitymode == 2:
            vgen = 0
            psi = 0
        else:
            vgen = self.Voltage
            psi = 0   
```